src/Features/CSharp/Portable/ValidateFormatString/ValidateFormatStringDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.ValidateFormatString;
+
+namespace Microsoft.CodeAnalysis.CSharp.ValidateFormatString
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class ValidateFormatStringDiagnosticAnalyzer : AbstractValidateFormatStringDiagnosticAnalyzer
+    {
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.InvocationExpression);
+        }
+
+        private static void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            var invocationExpr = (InvocationExpressionSyntax)context.Node;
+            if (invocationExpr.ArgumentList?.Arguments == null)
+            {
+                return;
+            }
+
+            SimpleNameSyntax name;
+            // When calling string.Format(...), Format will be of type MemberAccessExpressionSyntax 
+            if (invocationExpr.Expression is MemberAccessExpressionSyntax memberAccessExpr)
+            {
+                if (memberAccessExpr.Name?.ToString() != "Format")
```

Comment:
1. memberAccessExpr.Name cannot be null, so ?. is not necessary.  
2. Instead of calling .ToString(), call ".Identifier.ValueText".  
3. use "nameof(string.Format)".
4. Validate that the "arity" of "Name" is 0.